src/ngraph_builder.cc

```diff
@@ -2437,19 +2437,42 @@ static Status TranslateStridedSliceOp(
 
   std::shared_ptr<ng::Node> ng_strided_slice =
       make_shared<ng::op::Slice>(ng_input, l, u, s);
+
+  NGRAPH_VLOG(3) << " NG Lower Vector " << ng::join(lower_vec);
+  NGRAPH_VLOG(3) << " NG End Vector " << ng::join(end_vec);
+  NGRAPH_VLOG(3) << " NG Stride Vector " << ng::join(stride_vec);
+
   if (tf_shrink_axis_mask) {
-    NGRAPH_VLOG(3) << "shrink_axis_mask: " << tf_shrink_axis_mask;
-    NGRAPH_VLOG(3) << "shape of slice before axis shrinking: "
-                   << ng_strided_slice->get_shape();
+    set<int> shrink_axis;
+    int64 shrink_axis_mask = tf_shrink_axis_mask;
+
+    for (int i = 0; i < 64; i++) {
+      if ((shrink_axis_mask & 1) == 1) {
+        shrink_axis.insert(i);
+      }
+      shrink_axis_mask >>= 1;
+    }
+
+    NGRAPH_VLOG(3) << "Shrink axis mask " << tf_shrink_axis_mask;
+    NGRAPH_VLOG(3) << " Shrink axis " << ng::join(shrink_axis);
+
+    vector<size_t> output_shape;
+    for (int i = 0; i < lower_vec.size(); i++) {
+      if (shrink_axis.find(i) == shrink_axis.end()) {
+        output_shape.push_back(end_vec[i] - lower_vec[i]);
+      }
+    }
 
```

Comment:
The two loops can be combined:
for (int i = 0; i < lower_vec.size(); i++){
if ((shrink_axis_mask>>i & 1) == 1){
  output_shape.push_back(end_vec[i] - lower_vec[i]);
}
}

 If we feel 'shrink_axis' might be used later for other cases, then we might the 2 loop approach.

Reply:
yes. Thanks. I will update this.